tensorboard/plugins/core/core_plugin.py

```diff
@@ -389,11 +389,14 @@ def define_flags(self, parser):
         parser.add_argument(
             "--load_fast",
             type=str,
-            default="false",
+            default="auto",
             choices=["false", "auto", "true"],
             help="""\
-Experimental. Use a data server to accelerate loading. Set to "auto" to use a
-data server only if installed and supported for this invocation.
+Use alternate mechanism to load data. Typically 100x faster or more. Set to
```

Comment:
Wording: right now this is worded in a way that suggests to me as though simply passing the flag enables the alternate mechanism (like `store_true`), which is a bit confusing given that it's actually taking a value, and the default value is in fact `auto` rather than on or off.

Would recommend something more like

> Whether to use an accelerated mechanism to load data. Typically 100x faster or more, but only available on some platforms and for some invocations. Defaults to "auto" to use this new mode only if available (generally, if `tensorboard-data-server` is installed) and if supported for this invocation, otherwise falls back to the legacy loading path. Set to "true" to suppress the advisory note and hard-fail if the fast codepath is not available. Set to "false" to always fall back.

Also, maybe worth mentioning the GH issue for feedback here as well.

Reply:
Sounds good; thanks. Dropped “if `tensorboard-data-server` is installed”
to avoid confusion because the Python package is always installed (it
just might have `server_binary() is None`).

It’s also helpful that `argparse` shows the set of valid choices:

```
  --load_fast {false,auto,true}
                        Use alternate mechanism to load data. Typically 100x
                        faster or more, but only available on some platforms
                        and invocations. Defaults to "auto" to use this new
                        mode only if only if available, otherwise falling back
                        to the legacy loading path. Set to "true" to suppress
                        the advisory note and hard-fail if the fast codepath
                        is not available. Set to "false" to always fall back.
                        Feedback/issues:
                        https://github.com/tensorflow/tensorboard/issues/4784
                        (default: auto)
```

…so I think that the tri-state status is probably sufficiently clear.

Reply:
Oh right... hm. It would be nice if there was something slightly more concrete we could say here about when the fast path will be taken, since it feels fairly vague right now, but I guess now that we've outsourced that decision to the pip package or to the included server binary itself, we can't really do that without this potentially becoming stale.

Reply:
Agreed; I have wondered whether it will be confusing that `auto`
sometimes uses the fast path and sometimes doesn’t (sounds silly to put
it that way, since that’s its purpose, but I think the point stands).
I contemplated adding a `WARNING: Can't use fast path because`... when
`--load_fast=auto` is given (maybe only when given explictly?), but
worried that it would be too spammy for people who always use
incompatible features like `--logdir_spec` (deprecated) or unsupported
filesystems. Maybe it’s worth it? Open to suggestions.

Reply:
No particular suggestions.  I agree it would be easy for that to be spammy.  To some extent I think there's just a tension here between wanting a clean separation of concerns and wanting to be able to present the user with an enumeration of what is/isn't supported.  Coming down on the side of the former is probably fine and if it's a usage impediment we can revisit.